clusterloader2/pkg/measurement/common/service_creation_latency.go

```diff
@@ -0,0 +1,411 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package common
+
+import (
+	"fmt"
+	"sort"
+	"sync"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/equality"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/klog"
+
+	"k8s.io/perf-tests/clusterloader2/pkg/execservice"
+	"k8s.io/perf-tests/clusterloader2/pkg/measurement"
+	measurementutil "k8s.io/perf-tests/clusterloader2/pkg/measurement/util"
+	"k8s.io/perf-tests/clusterloader2/pkg/measurement/util/informer"
+	"k8s.io/perf-tests/clusterloader2/pkg/measurement/util/runtimeobjects"
+	"k8s.io/perf-tests/clusterloader2/pkg/measurement/util/workerqueue"
+	"k8s.io/perf-tests/clusterloader2/pkg/util"
+)
+
+const (
+	serviceCreationLatencyName           = "ServiceCreationLatency"
+	serviceCreationLatencyWorkers        = 10
+	defaultServiceCreationLatencyTimeout = 1 * time.Minute
+	defaultWaitTimeout                   = 10 * time.Minute
+	defaultCheckInterval                 = 10 * time.Second
+	pingBackoff                          = 1 * time.Second
+	pingChecks                           = 10
+)
+
+func init() {
+	measurement.Register(serviceCreationLatencyName, createServiceCreationLatencyMeasurement)
+}
+
+func createServiceCreationLatencyMeasurement() measurement.Measurement {
+	return &serviceCreationLatencyMeasurement{
+		queue:         workerqueue.NewWorkerQueue(serviceCreationLatencyWorkers),
+		creationTimes: make(map[string]*serviceCreationTimes),
+		pingCheckers:  make(map[string]*pingChecker),
+	}
+}
+
+type serviceCreationLatencyMeasurement struct {
+	namespace       string
+	labelSelector   string
+	fieldSelector   string
```

Comment:
There are already couple measurements, that are using both namespace, labelSelector and fieldSelector.
Can we opaque those 3 into some type (like objectSelector?) and add a common parsing method for those to avoid that boilerplate everywhere.

Reply:
Done.